pkg/asset/installconfig/azure/validation.go

```diff
@@ -77,6 +78,11 @@ func ValidateInstanceType(client API, fieldPath *field.Path, region, instanceTyp
 				errMsg := fmt.Sprintf("instance type does not meet minimum resource requirements of %d vCPUsAvailable", req.minimumVCpus)
 				allErrs = append(allErrs, field.Invalid(fieldPath.Child("type"), instanceType, errMsg))
 			}
+		} else if strings.EqualFold(*capability.Name, "HyperVGenerations") {
+			generation := strings.ReplaceAll(strings.ToUpper(to.String(capability.Value)), " ", "")
+			if !sets.NewString(strings.Split(generation, ",")...).Has("V1") {
```

Comment:
Let's use `strings.TrimSpace` instead of `strings.ReplaceAll` so that it is clearer on read what the intention is.
```suggestion
			generations := sets.NewString()
			for _, g := range strings.Split(to.String(capability.Value), ",") {
				g = strings.TrimSpace(g)
				g = strings.ToUpper(g)
				generations.Insert(g)
			}
			if !generations.Has("V1") {
```